# Profitable App Profiles for App Store and Google Play Markets

Our aim in this project is to find mobile app profiles that are profitable for the App Store and Google Play markets and enable our team of developers to make data-driven decisions with respect to the kind of apps they build.

At our company, we only build apps that are free to download and install, and our main source of revenue consists of in-apps ads. This means that our revenue for any given apps is mostly influenced by the number of users use our apps. Our goal of this project is to analyze data to help our developers understand what kinds of apps are more likely to attract more customers from 10,000 Android Apps and 7,000 iOS apps.

# Opening and Exploring Data

As of September 2018, there were approximately 2 million iOS apps available on App Store, and 2.1 million Android apps on Google Play. 
![image](https://s3.amazonaws.com/dq-content/350/py1m8_statista.png) Source : [Statista](https://www.statista.com/statistics/276623/number-of-apps-available-in-leading-app-stores/)

Collecting data for over four million apps requires a signnificant amount of time and money, so we'll try to analyze a sample of data instead. To avoid speading resources to collect new data ourselves, we should first try to see whether we can find any relevant existing data at no cost. Luckily, there are two data sets that seem suitable for our purpose:
-  A data set containing data about approximately 10,000 Android apps from Google Play. You can download the data set directly from this link.
-  A data set containing data about approximately seven thousand 7,000 iOS apps from the App Store. You can download the data set directly from this link.

Let's start by opening the two data sets and then continue with exploring the data.

In [144]:
from csv import reader

### The Google Play data set ###
opended_file = open ('googleplaystore.csv')
read_file = reader(opended_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

### The App Store data set ###
opended_file = open('AppleStore.csv')
read_file = reader (opended_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]


To make it easier to explore the data sets, we'll first write a function named `explore_data()` that we can use repeatedly to explore rows in a more readable way. We'll also add an option for our function to show the number of rows and colums to any data set.

In [145]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slide = dataset[start:end]
    for row in dataset_slide:
        print(row)
        print('\n') #add new empty row
        
        
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))
        
print(android_header)    
print('\n')
explore_data(android, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


We see that the Google Play data set has 10841 apps and 13 columns. At first glance, the columns that might be useful in our analysis are `'App'`, `'Category'`, `'Reviews'`, `'Installs'`, `'Type'`, `'Price'`, and `'Genres'`.

Now we'll look at the App Store data set:

In [25]:
print(ios_header)
print('\n')
explore_data(ios,0 ,3 ,True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of colums: 16


The App Store has 7197 apps and 16 columns. The columns that can be used in the analysis are `'track_name'`, `'price'`, `'user_rating'`, `'cont_rating'`, `'prime_genre'`. Not all column are self-explanatory in this case, but details about each column can be found in [this documentation](https://www.kaggle.com/datasets/ramamet4/app-store-apple-data-set-10k-apps).

# Deleting Wrong Data 

The Google Play data set has a discussion section, and we can see that one of the [discussions](https://www.kaggle.com/datasets/ramamet4/app-store-apple-data-set-10k-apps/discussion/106176) outlines an error for row 10472. Let's print this row and compare this against the header and another row that is correct.

In [27]:
print (android[10472]) #the wrong row
print('\n')
print (android_header) #the header row
print('\n')
print (android[3]) #the correct row

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


The row 10472 corresponds to the app Life Made WI-Fi Touchscreen Photo Frame, and we can see that the rating is 19. This is clearly off because the maximum rating for a Google Play app is 5 (as mentioned in the discussions section, this problem is caused by a missing value in the 'Category' column). As a consequence, we'll delete this row.

In [28]:
print(len(android))
del android[10472] #only do this once
print(len(android))

10841
10840


# Removing Duplicate Entries

After looked at the Google Play data set, we find that some apps have more than one entry. For instance, the application Instagram has four entries:

In [32]:
for app in android:
    name = app[0]
    if name == 'Instagram':
        print(app)
 

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In total, there are 1,181 apps occur more than once:

In [37]:
duplicate_apps = []
unique_apps = []

for app in android:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(app[0])
    else:
        unique_apps.append(app[0])
        
print('Number of duplicate apps:',len(duplicate_apps))
print('\n')
print ('Example of duplicate apps:', duplicate_apps[:15])

Number of duplicate apps: 1181


Example of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


We don't want to count certain apps more than once when we analyze data, so we need to remove the duplicate entries and keep only one entry per app.

If we examine the rows we printed two cells above  for the Instagram app, the main difference happend on the  fourth position of each row, which is the number of reviews. The different numbers show that the data was collected  at different time. We can use this to build a criterion for keeping rows. We don't remove rows randomly, but rather we''ll keep the rows that have the highest number of reviews because the higher the number of reviews, the more reliable the ratings.

To do that, we  will:
* Create a dictionary where each key is a unique app name, and the value is the highest number of reviews of that app.
* Use the dictionary to create a new data set, which will have only one entry per app(and we only select the apps with the highest number of reviews).

# Clear Data Set

In [43]:
reviews_max = {}

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:    
        reviews_max[name] = n_reviews


In a previous code cell, we found that there are 1,181 cases where an app occurs more than once, so the length of our dictionary (of unique apps) should be equal to the difference between the length of our data set and 1,181.

In [48]:
print ('Expected length:', len(android) - 1181)
print ('Actual length:', len(reviews_max))

Expected length: 9659
Actual length: 9659


Now, let's use the `reviews_max` dictionary to remove the duplicates. For the duplicate cases, we'll only keep the entries with the highest number of reviews . In the code below:
* Initializing two empty lists, `android_clean` and `already_added`.
* We loop through the `android` data set, for ever iteration:
    - We isolate the name of the app and the number of reviews.
    - We add the current `app` row to the `android_clean` list, and the app name(`name`) to the `aldready_added` list. If:
         * The number of reviews of the current app matches the number of reviews of that app as described in the `reviews_max` dictionary; and
         * The name of the app is not already in the `already_added` list. We need to add this supplementary condition to account for those cases where the highest number of reviews of a duplicate app is the same for more than one entry (for example, the Box app has three entries, and the number of reviews is the same). If we just check for `reviews_max[name] == n_reviews`, we'll still end up with duplicate entries for some apps.

In [64]:
android_clean = []
already_added = []

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if (name not in already_added) and (reviews_max[name] == n_reviews):
        android_clean.append(app)
        already_added.append(name) #have to be inside the if block


Now, let's quickly explore the new data set, and confirm that the number of rows is 9,659

In [65]:
explore_data(android_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of colums: 13


# Removing Non-English Apps

## Part One

The names of some of the apps suggest they are not directed toward an English-speaking audience. Below are some couple of examples from both data sets:

In [70]:
print(ios[813][1])
print(ios[6731][1])

print(android_clean[4412][0])
print(android_clean[7940][0])

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜
中国語 AQリスニング
لعبة تقدر تربح DZ


We don't need these kind of apps, so we'll remove them. We'll use symbol used in English text, which includes letters from the English alphabet, numbers composed of digits from 0 to 9, punctuation marks(., !, ?, ;, etc.) and other symbols (+, *, /, etc.). 

All these characters that are specific to English texts are encoded using the ASCII standard. Eacj ASCII character has a corresponding number between 0 and 127 associated with it, and we can takw advantage of that to build a function that checks an app name and tells us whether it contains non-ASCII characters.

We built this function below, and we use thw built-in function `ord()` to find out yhe corresponding encoding number of each character.

In [73]:
def is_english(string):
    
    for character in string:
        if ord(character) > 127:
            return False
    return True

print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))

True
False


## Part Two

The function seems to work fine, but some English app names use emojis or other symbols (™, — (em dash), – (en dash), etc.) that fall outside of the ASCII range. Because of this, we'll remove useful apps if we use the function in its current form.

In [74]:
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

print(ord('™'))
print(ord('😜'))

False
False
8482
128540


To minimize the data loss, we only remove an app that its name has more than three non-ASCII characters:

In [78]:
def is_english(string):
    non_ascii = 0
    
    for character in string:
        if ord(character) > 127:
            non_ascii += 1
    
    if non_ascii > 3:
        return False
    else:
        return True
    
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))       

True
True


The function is still not perfect, but it seems good at this point in our analysis - we shouldn's spend too much time on optimization at this point.

Below, the functions `is_english` is used to filter both data sets:

In [81]:
android_english = []
ios_english = []

for app in android_clean:
    name = app[0]
    if is_english(name) :
        android_english.append(app)

for app in ios:
    name = app[1]
    if is_english(name):
        ios_english.append(app)
        
explore_data(android_english, 0, 3, True)
print('\n')
explore_data(ios_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of colums: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', '

We can see that we're left with 9614 Android apps and 6183 iOS apps


# Isolating the free apps

As we mentioned in the introduction, we only build apps that are free to download and install, and our main source of revenue consists of in-apps ads. Our data sets contain both free and non-free apps, and we'll need to seperate the free apps for our analysis. Below, we isolate the free apps for both our data sets.

In [87]:
android_final = []
ios_final = []

for app in android_english:
    price = app[7]
    if price =='0':
        android_final.append(app)
        
for app in ios_english:
    price = app[4]
    if price == '0.0':
        ios_final.append(app)
        
print('Number of Android free apps:', len(android_final))
print('Number of iOS free apps:', len(ios_final))

Number of Android free apps: 8864
Number of iOS free apps: 3222


# Most Common Apps by Genre

## Part One

As we mentioned in the introduction, our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of users.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

1. Build a minimal Android version of the app, and add it to Google Play.
1. If the app has a good response from users, we then develop it further.
1. If the appis profitable after six months, we also build an iOS version of the app and add it to the App Store.

Becasue our end goal is to add the app on both the App Store and Google Playm we need to find app profiles that are successsful on both markets. For instance, a profile that might work well for both markets might be a productivity app that makes use of gamification.

Let's begin the analysis by getting a sense of the most common genres for each market. For this, we'll build a frequency table for the `prime_genre` column of the App Store data set , and the `Genre` and `Category` colums of the Google Play data set.

## Part Two

We'll build two functions we can use to analyze the frequency tables:

 * One function to generate frequency tables that show percentages
 * Another functions that we can use to display the percentages in a descending order
    

In [99]:
def freq_table(dataset, index):
    table = {}
    total = 0
    for row in dataset:
        value = row[index]
        total += 1
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
            
    table_percentage = {}
    for key in table:
        percentage = (table[key]/total) * 100
        table_percentage[key] = percentage
        
    return table_percentage    


def display_table(dataset, index): 
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_as_tuple = (table[key], key)  # transform the frequency table into a list of tuple
        table_display.append(key_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True) # print percentages in a descending order
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

## Part Three

We'll examining the freqency table of `prime_genre` colums of App Store data set

In [100]:
display_table(ios_final, -5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


  We can see that among the free English apps, more than a half (58.16%) are games. Entertainment apps are close to 8%, followed by photo and video apps, which are close to 5%. Only 3.66% of the apps are designed for education, followed by social networking apps which amount for 3.29% of the apps in our data set.
        
   The general impression is that App Store (at least the part containning free English apps) are dominated by apps that are designed for entertaining purposes (games, entertainment, photo and video, social networking, sports, music, etc.), while apps with pratical purposes (education, shopping, ultilies, productivity, lifestyle, etc.) are more rare. However, the fact that entertaining apps that are the numerous does not also amply that they are also have the greatest number of users - the demand might not be the same as the offer.
    
       
   We continue to examinate the `Genre` and `Category` columns of the Google Play data set (two columns which seem to be related).

In [101]:
display_table(android_final, 1) # Category

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

 The landscape seems significant different on Google Play: there are not that many apps for entertaining purposes, and it seems that a good number of apps are designed for practical purposes (family, tools, business, lifestyle, productivity, etc.). However, if we investigate this further, we can see that the family category (which almost 19% of the apps)means mostly games for kids

[![](https://camo.githubusercontent.com/0d974d86dfcb791ca1d1505f810ba5a191bb6248e413d9b6ae1a60b211cca918/68747470733a2f2f73332e616d617a6f6e6177732e636f6d2f64712d636f6e74656e742f3335302f7079316d385f66616d696c792e706e67)](https://play.google.com/store/apps/category/FAMILY?hl=en)

 Even so , practical apps seem to have a better representation on Google Play compared to App Store. This picture is also confirmed by the frequency table we see for the `Genre` column:

In [103]:
display_table(android_final, -4)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

The difference between `Genre` and `Category` columns is not crystal clear, but one thing we can notice is that the `Genre` column has more variable (it has more categories). We're only looking for the bigger picture at the moment, so we'll only qork with the `Category` column moving forward.

UP to this point, we found that the App Store is dominated by apps designed for entertaining purposes, while Google Play shows a more balanced landscape of both practical and entertaining purposes apps. Now. we'd like to get an idea about the kind of apps have most users.

# Most Popular Apps by Genre on App Store


One way to know what genres are the most popular (have the most users) is to calculate the avarage number of installs for each app genre. For the Google Play data set, we can find this information in the `Installs` column, but for the App Store data set this information is missing. Therefore, we'll take the total number of user ratings as a proxy, which we can find in the `rating_count_tot` column.

Below, we calculate the average number of users ratings per app genre on the App Store:

In [108]:
genre_ios = freq_table(ios_final, -5)

for genre in genre_ios:
    total = 0      # sum of the users rating
    len_genre = 0  # sum of number of apps
    for app in ios_final:
        genre_app = app[-5]
        if genre_app == genre:
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings) 

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


On average, Navigation apps have the highest number of user reviews, but this firgure is heavily influenced by Waze and Google Maps, which have close to half a million user reviews together:

In [110]:
for app in ios_final:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5])  #print name and number of ratings

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


The same patter applies to social networking apps, where the average number is heavily influenced by a few giants like Facebook, Pinterest, Skype, etc. Same applies to music apps, where a few big players like Pandora, Sportify, and Shazam heavily influence the average number.

Our aim is to find popular genres, but navigation, social networking and music apps might seem more popular than they really are. The average number of ratings seem to be skewed by vey fews apps which have hundreds of thousands of user ratings, while the other apps may struggle to get past the 10,000 threhold. We could get a better picture by removing these extremely popular apps for each genre and then rework the averages, but we'll leave this detail for later.

Reference apps have 74,942 user ratings on average, but it's actually the Bible and Dictionary.com which skew up the average rating:


In [111]:
for app in ios_final:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


However, this niche seems to show some potential. One thing we could do is take another popular book and turn it into an app where we could add different features besides the raw version of the book.

This idea seems to fit well with the fact that the App Store is dominated by entertaining purposes apps. This also suggests that the market right now might be a bit saturated with for-fun apps, which means a practical app might have more of a chance to stand out among the huge number of apps on the App Store.

Other genres that seem popular include weather, book, food and drink, or finance. The book genre seem to overlap a bit with the app idea we described above, but the other genres don't seem too interesting to us:

* Weather apps — people generally don't spend too much time in-app, and the chances of making profit from in-app adds are low. Also, getting reliable live weather data may require us to connect our apps to non-free APIs.

* Food and drink — examples here include Starbucks, Dunkin' Donuts, McDonald's, etc. So making a popular food and drink app requires actual cooking and a delivery service, which is outside the scope of our company.

* Finance apps — these apps involve banking, paying bills, money transfer, etc. Building a finance app requires domain knowledge, and we don't want to hire a finance expert just to build an app.

Now let's analyze the Google Play market a bit.


# Most Popular Apps by Genre on Google Play

For the Google Play data set, we have the number of installs. So, we should be able to get a clearer picture about genre popularity. 

However, one problem with this data is that it is not precise.


In [112]:
display_table(android_clean, 5) # the Installs columns

1,000,000+ : 14.670255720053834
100,000+ : 11.522931980536287
10,000+ : 10.663629775339063
10,000,000+ : 9.700797183973496
1,000+ : 9.183145253131794
100+ : 7.340304379335334
5,000,000+ : 6.284294440418263
500,000+ : 5.217931462884357
50,000+ : 4.855575111295165
5,000+ : 4.8452220726783315
10+ : 3.985919867481106
500+ : 3.3957966663215657
50,000,000+ : 2.112019877834144
50+ : 2.112019877834144
100,000,000+ : 1.9567242985816338
5+ : 0.8489491665803914
1+ : 0.6936535873278807
500,000,000+ : 0.24847292680401697
1,000,000,000+ : 0.20706077233668083
0+ : 0.14494254063567658
0 : 0.01035303861683404


This data is open-ended. For instance, we don't know whether an app with 100,000+ installs has 100,000 installs, 200,000, or 300,000. However, we only need to know which app genres have the most users, so we don't need to find perfect precision with respect to the number to the number of users.

We're going take the number as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 intalls, and an app with 1,000,000+ installs has 1,000,000 installs, and so on.

And in teh computation, we convert each install number to `float` to avoid error. 
 * Compute the average number of installs for each genre (category).

In [118]:
genre_android = freq_table(android_clean, 1)

for category in genre_android:
    total = 0
    len_category = 0
    
    for app in android_clean:
        category_app = app[1]
        if category_app == category:
            n_installs = app[5]
            n_installs = n_installs.replace('+', '')
            n_installs = n_installs.replace(',', '')
    
            total += float(n_installs)
            len_category += 1
    
    avg_installs = total / len_category
    print(category, ':', avg_installs)
            

ART_AND_DESIGN : 1856362.2950819673
AUTO_AND_VEHICLES : 625061.305882353
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 7504367.459459459
BUSINESS : 1659916.3452380951
COMICS : 803234.8214285715
COMMUNICATION : 35042146.82857143
DATING : 828971.2176470588
EDUCATION : 1775252.3364485982
ENTERTAINMENT : 11375402.298850575
EVENTS : 249580.640625
FINANCE : 1319851.4028985507
FOOD_AND_DRINK : 1891060.2767857143
HEALTH_AND_FITNESS : 3972300.388888889
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 630903.6904761905
LIFESTYLE : 1365375.4444444445
GAME : 14226135.745243128
FAMILY : 3319926.0965848453
MEDICAL : 96944.49873417722
SOCIAL : 22961790.384937238
SHOPPING : 6932419.727722772
PHOTOGRAPHY : 16577037.918149466
SPORTS : 3373767.6861538463
TRAVEL_AND_LOCAL : 13218662.767123288
TOOLS : 9774151.887816647
PERSONALIZATION : 4075783.994680851
PRODUCTIVITY : 15489549.114973262
PARENTING : 525351.8333333334
WEATHER : 4570892.658227848
VIDEO_PLAYERS : 23975016.585365854
NEWS_AND_MAGAZI

On average, communication apps have the most installs: 28,456,119. However, this number is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs:

In [123]:
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                     or app[5] == '500,000,000+'
                                     or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])
    

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

In [124]:
under_100_m = []

for app in android_final:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'COMMUNICATION') and (float(n_installs) < 100000000):
        under_100_m.append(float(n_installs))
        
sum(under_100_m)/len(under_100_m)        

3603485.3884615386

When we removed all the communication apps that have over 100 million installs, the average would be reduced roughly ten times.

The same pattern applies for `video players` (dominated by Youtube, Google Play Movies & TV, or MX Player), `social` (Facebook, Instagram, Google+, etc.), `photography`(Google Photos, B612 and other popular photo editors), and `productivity` (Google Drive,Microsoft Word, Dropbox,Evernote, etc.) apps.

In [133]:
for app in android_final:
    if app[1] == 'VIDEO_PLAYERS'and (app[5] == '1,000,000,000+'
                                     or app[5] == '500,000,000+'
                                     or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

YouTube : 1,000,000,000+
Motorola Gallery : 100,000,000+
VLC for Android : 100,000,000+
Google Play Movies & TV : 1,000,000,000+
MX Player : 500,000,000+
Dubsmash : 100,000,000+
VivaVideo - Video Editor & Photo Movie : 100,000,000+
VideoShow-Video Editor, Video Maker, Beauty Camera : 100,000,000+
Motorola FM Radio : 100,000,000+


In [134]:
for app in android_final:
    if app[1] == 'SOCIAL'and (app[5] == '1,000,000,000+'
                                     or app[5] == '500,000,000+'
                                     or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Facebook : 1,000,000,000+
Facebook Lite : 500,000,000+
Tumblr : 100,000,000+
Pinterest : 100,000,000+
Google+ : 1,000,000,000+
Badoo - Free Chat & Dating App : 100,000,000+
Tango - Live Video Broadcast : 100,000,000+
Instagram : 1,000,000,000+
Snapchat : 500,000,000+
LinkedIn : 100,000,000+
Tik Tok - including musical.ly : 100,000,000+
BIGO LIVE - Live Stream : 100,000,000+
VK : 100,000,000+


In [132]:
for app in android_final:
    if app[1] == 'PHOTOGRAPHY'and (app[5] == '1,000,000,000+'
                                     or app[5] == '500,000,000+'
                                     or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

B612 - Beauty & Filter Camera : 100,000,000+
YouCam Makeup - Magic Selfie Makeovers : 100,000,000+
Sweet Selfie - selfie camera, beauty cam, photo edit : 100,000,000+
Google Photos : 1,000,000,000+
Retrica : 100,000,000+
Photo Editor Pro : 100,000,000+
BeautyPlus - Easy Photo Editor & Selfie Camera : 100,000,000+
PicsArt Photo Studio: Collage Maker & Pic Editor : 100,000,000+
Photo Collage Editor : 100,000,000+
Z Camera - Photo Editor, Beauty Selfie, Collage : 100,000,000+
PhotoGrid: Video & Pic Collage Maker, Photo Editor : 100,000,000+
Candy Camera - selfie, beauty camera, photo editor : 100,000,000+
YouCam Perfect - Selfie Photo Editor : 100,000,000+
Camera360: Selfie Photo Editor with Funny Sticker : 100,000,000+
S Photo Editor - Collage Maker , Photo Collage : 100,000,000+
AR effect : 100,000,000+
Cymera Camera- Photo Editor, Filter,Collage,Layout : 100,000,000+
LINE Camera - Photo editor : 100,000,000+
Photo Editor Collage Maker Pro : 100,000,000+


In [131]:
for app in android_final:
    if app[1] == 'PRODUCTIVITY'and (app[5] == '1,000,000,000+'
                                     or app[5] == '500,000,000+'
                                     or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Microsoft Word : 500,000,000+
Microsoft Outlook : 100,000,000+
Microsoft OneDrive : 100,000,000+
Microsoft OneNote : 100,000,000+
Google Keep : 100,000,000+
ES File Explorer File Manager : 100,000,000+
Dropbox : 500,000,000+
Google Docs : 100,000,000+
Microsoft PowerPoint : 100,000,000+
Samsung Notes : 100,000,000+
SwiftKey Keyboard : 100,000,000+
Google Drive : 1,000,000,000+
Adobe Acrobat Reader : 100,000,000+
Google Sheets : 100,000,000+
Microsoft Excel : 100,000,000+
WPS Office - Word, Docs, PDF, Note, Slide & Sheet : 100,000,000+
Google Slides : 100,000,000+
ColorNote Notepad Notes : 100,000,000+
Evernote – Organizer, Planner for Notes & Memos : 100,000,000+
Google Calendar : 500,000,000+
Cloud Print : 500,000,000+
CamScanner - Phone PDF Creator : 100,000,000+


Again the main concern is that these app genres might seem more popular than they really are. Moreover, these niches seem to be dominated by a few giants who are hard to compete against.

The game genre seems pretty popular, but previously we found out this part of the market seeems a bit saturated.

The books and reference genre looks fairy popular as well, with an average of installs of 8,767,811. It's interesting to explore this in more depth, since we found this genre has some potential to work well on the App Store, and our aim is to recommend an app genre that shows potential for being profitable on both the App Store and Google Play.

Let's take a look at some of the apps from this genre and their number of installs:

In [141]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

The books and reference genre incluse a variety of apps: software for processiong and reading ebooks. various collections of libaries, dictionsries, tutorials on programming or languages, etc. 

It seems that there's still a number of extremely popular apps that skew the average:

In [142]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


However, it looks like there are only a few very popular apps, so this market still shows potential. So we get some app ideas based on the kind of apps that are somewhere in the middle in terms of popularity (between 1,000,000 and 100,000,000 downloads).

In [143]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

There are quite a few apps that built around the book Al-Quran, which suggests that building an app around a popular book can be profitable. It seems that taking a popular book (perhaps a more recent book) and turning it into an app could be profitalbe for both the Google Play and App Store markets.

however, it looks like the market is already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the books, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.

# Key Insights

In this project, we analyzed data about the App Store and Google Play mobile apps with respect to the goal of recommending an app profiles that can be profitable for both markets.

1. Turning popular books (and more recent books) into an app could be profitable for both the Google Play and the App Store markets.

2. The markets are already full of libraries, so we need to add some special features besides the raw version of the book:

a. Daily quotes form the book

b. An audio version of the books

c. Quizzes on the books

d. A forum where people can discuss the book
